cli/cmd/component-delete.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/kinvolk/lokomotive/pkg/components"
+	"github.com/kinvolk/lokomotive/pkg/components/util"
+	"github.com/kinvolk/lokomotive/pkg/config"
+)
+
+var componentDeleteCmd = &cobra.Command{
+	Use:   "delete",
+	Short: `Delete a component installed in the cluster.`,
+	Run:   runDelete,
+}
+
+//nolint:gochecknoinits
+func init() {
+	componentCmd.AddCommand(componentDeleteCmd)
+}
+
+func runDelete(cmd *cobra.Command, args []string) {
+	contextLogger := log.WithFields(log.Fields{
+		"command": "lokoctl component delete",
```

Comment:
Did we agree between the following:
- delete (kubectl)
- destroy (terraform)
- remove
- uninstall (helm)
?

Reply:
Nobody raised a concern on using delete, if there is a concern then we can discuss I think.

Reply:
We have `cluster destroy`, so for consistency, I would go for `component destroy` too.

Reply:
I agree that this will be consistent with the cluster destroy. But then it does not feel intuitive. Now we need to find the right tradeoff whether we want consistency with `terraform` or with `kubectl`.

@iaguis @johananl @joaquimrocha  your opinions will be of help here?

Reply:
We decided on the call for `delete`.